api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/AvgPool1D.kt

```diff
@@ -0,0 +1,95 @@
+/*
+* Copyright 2020 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+* Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+*/
+
+package org.jetbrains.kotlinx.dl.api.core.layer.pooling
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.layer.convolutional.ConvPadding
+import org.jetbrains.kotlinx.dl.api.core.shape.convOutputLength
+import org.jetbrains.kotlinx.dl.api.inference.keras.CHANNELS_FIRST
+import org.jetbrains.kotlinx.dl.api.inference.keras.CHANNELS_LAST
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+import org.tensorflow.op.core.Squeeze
+
+/**
+ * Average pooling operation for 1D temporal data (e.g. audio, timseries).
+ *
+ * Downsamples the input by taking the average over a temporal window of size [poolSize].
+ *
+ * @property [poolSize] Size of the temporal pooling window.
+ * @property [strides] The amount of shift for pooling window in each pooling step. If
+ * `null`, it will default to [poolSize].
+ * @property [padding] Padding strategy; can be either of [ConvPadding.VALID] which means no
+ * padding, or [ConvPadding.SAME] which means padding the input equally such that the output
+ * has the same dimension as the input.
+ * @property [dataFormat] Data format of input; can be either of [CHANNELS_LAST] or [CHANNELS_FIRST].
+ */
+public class AvgPool1D(
+    public val poolSize: Int = 2,
+    public val strides: Int? = null,
```

Comment:
Could we add here the default value?

Reply:
Since in pooling layers it's very common for strides to be the same as pool size value, the benefit of using `null` as default value is that the user won't need to set it explicitly and we just default to pool size value instead. This is the same behavior in Keras and PyTorch.

Reply:
Kotlin and Python users have a different experience with nullability, and Kotlin is trying to be a place without nulls.
As I commented in the MaxPool1D layer PR, please change it to the default value 1,2,1.

We need to keep control of the parameters that could be a hyper-parameter for AutoML tasks and show it to the user.
As a variant - to make a warning if stride and pool are not equal with reminder that it's a common practice

Reply:
I see. I understand your point about removing nullability and hidden dependencies. However, I don't get the part about losing control of the parameter during hyper-parameter tuning scenarios or architecture search. Anyways, since I am new to Kotlin I think you have a much better understanding of what would be the best decision here and I would follow it gladly :)

Reply:
I meant that we train the user to use only one parameter and leave the second **null** to change together. Perhaps, looking for a new architecture, a method a la **NAS**, something with this approach will be missed, at least I think so.